### DISCUSSION TITLE
Possibility to flash a device (e.g. FC) connected to the receiver via wifi (or wifi-passthru)

### DISCUSSION THREAD

**Discussion by wimalopaan** [ExpressLRS/ExpressLRS#3061] [Ideas]
It would be great to have the possibility to flash a device connected to the receiver. This device is normally a kind-of flight-controller or similar or some kind of crsf-switch, connected via full-duplex serial to the receiver. 
Flashing this device needs dismounting (at least disconnecting). It would be much easier to flash that device via wifi.

The receiver does not use crsf protocol in the moment, it is in wifi mode. So, it would be possible, to use some simple file transfer (like e.g. zmodem) protocol. 
Another way would be to define some crsf commands (0x32) to transfer chunks of the file to the device.

**Comment by Mkorho** [comment]
Adds MSP to CRSF over TCP (WIFI) feature #1274

ELRS Wifi Betaflight Configurator support was added in v3.0.
If you want more features to it, you need to talk to Betaflight team.

**Comment by wimalopaan** [comment]
Oh, cool, #1274 looks very promising.

Is there any information about that (other than the source)?
If I understand that correct, all MSP packets are inserted into the packet stream as if they where received over the air? And backwords the same?

**Comment by wimalopaan** [comment]
Is it possible to insert raw CRSF packets into the packet stream over wifi? This would add the possibility to use e.g. ELRSBuddy over wifi.
